frontend/amundsen_application/static/js/pages/HomePage/index.tsx

```diff
@@ -29,6 +24,43 @@ export interface DispatchFromProps {
 
 export type HomePageProps = DispatchFromProps & RouteComponentProps<any>;
 
+const getHomePageWidgetComponents = (
+  layout: HomePageWidgetsConfig
+): React.ReactNode[] => {
+  /* Looks up each Widget in layout by its name property, and if found, 
+  puts the relevant component's JSX into the output array. */
+
+  const res: React.ReactNode[] = [];
+
+  layout.widgets.forEach((widget) => {
+    const WidgetComponent = React.lazy(
+      () =>
+        import('/js/features/HomePageWidgets/' + widget.options.path + '.tsx')
+    );
+
+    const additionalProps = widget.options.additionalProps
+      ? widget.options.additionalProps
+      : null;
+
+    res.push(
+      <div className="home-element-container">
+        <React.Suspense fallback={<div>Loading...</div>}>
+          <WidgetComponent {...additionalProps} />
+        </React.Suspense>
+      </div>
+    );
+  });
+
+  return res;
```

Comment:
I would do just a map here.
.forEach and .push are methods that we barely use in nowadays' JavaScript, as they imply mutations of data. We prefer to create new data with .map